core/comm/client.go

```diff
@@ -213,3 +213,37 @@ func (client *GRPCClient) NewConnection(address string, tlsOptions ...TLSOption)
 	}
 	return conn, nil
 }
+
```

Comment:
any reason this is not in creds.go where we have the analog to this for the server?

Reply:
Will move it

Reply:
moved it